src/mqt/benchviewer/backend.py

```diff
@@ -23,256 +23,544 @@
+                        & (db_tmp["compiler"] == "qiskit")
+                        & (db_tmp["compiler_settings"] == opt_lvl)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp5])
+
+        if mapped_tket_compiler:
+            for placement in mapped_tket_placements:
+                for device in mapped_devices:
+                    db_tmp6 = db_tmp.loc[
+                        (db_tmp["mapped_flag"])
+                        & (db_tmp["target_device"] == device)
+                        & (db_tmp["compiler"] == "tket")
+                        & (db_tmp["compiler_settings"] == placement)
+                    ]
+                    db_filtered = pd.concat([db_filtered, db_tmp6])
+
+        return db_filtered["path"].to_list()
+
+    class NoSeekBytesIO:
+        def __init__(self, fp: io.BytesIO):
+            self.fp = fp
+            self.deleted_offset = 0
+
+        def write(self, b):
+            return self.fp.write(b)
+
+        def tell(self):
+            return self.deleted_offset + self.fp.tell()
+
+        def hidden_tell(self):
+            return self.fp.tell()
+
+        def seekable(self):
+            return False
 
-def get_opt_level(filename: str):
-    """Extracts the optimization level based on a filename.
-
-    Keyword arguments:
-    filename -- filename of a benchmark
-
-    Return values:
-    num -- optimization level
-    """
-
-    pat = re.compile(r"opt\d")
-    m = pat.search(filename)
-    num = m.group()[-1:] if m else -1
-    return int(num)
-
-
-def get_num_qubits(filename: str):
-    """Extracts the number of qubits based on a filename.
-
-    Keyword arguments:
-    filename -- filename of a benchmark
-
-    Return values:
-    num -- optimization level
-    """
-
-    pat = re.compile(r"(\d+)\.")
-    m = pat.search(filename)
-    num = m.group()[0:-1] if m else -1
-    return int(num)
+        def hidden_seek(self, offset, start_point=io.SEEK_SET):
+            return self.fp.seek(offset, start_point)
+
+        def truncate_and_remember_offset(self, size):
+            self.deleted_offset += self.fp.tell()
+            self.fp.seek(0)
+            return self.fp.truncate(size)
+
+        def get_value(self):
+            return self.fp.getvalue()
+
+        def close(self):
+            return self.fp.close()
+
+        def read(self):
+            return self.fp.read()
+
+        def flush(self):
+            return self.fp.flush()
+
+    def generate_zip_ephemeral_chunks(
+        self,
+        filenames: list[str],
+    ) -> Iterable[bytes]:
+        """Generates the zip file for the selected benchmarks and returns a generator of the chunks.
+
+        Keyword arguments:
+        paths -- list of file paths for all selected benchmarks
+
+        Return values:
+            Generator of bytes to send to the browser
+        """
+        fileobj = self.NoSeekBytesIO(io.BytesIO())
+
+        paths = [Path(name) for name in filenames]
```

Comment:
It shouldn't be necessary to create a whole list of all filenames here.
Just create the `Path` object on the fly down below.

Reply:
Good point, implemented that.